src/overlays/actors/ovl_Obj_Aqua/z_obj_aqua.c

```diff
@@ -26,55 +35,266 @@ const ActorInit Obj_Aqua_InitVars = {
+        }
+    } else if (this->counter <= 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBDCC.s")
+void func_80ACBD34(ObjAqua* this) {
+    this->actionFunc = func_80ACBD48;
+}
 
+void func_80ACBD48(ObjAqua* this, GlobalContext* globalCtx) {
+    if ((AQUA_HOT(&this->actor) == 1) && (this->alpha > 90)) {
+        func_80ACB940(this, globalCtx);
+    }
+    if (this->alpha > 5) {
+        this->alpha -= 5;
+    } else {
+        this->alpha = 0;
+    }
+    if (this->actor.shape.shadowAlpha > 2) {
+        this->actor.shape.shadowAlpha -= 2;
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80ACBDCC(ObjAqua* this) {
+    this->actor.gravity = -0.1f;
+    this->counter = 40;
+    this->alpha = 140;
+    this->actionFunc = func_80ACBDFC;
+}
+
+#ifdef NON_MATCHING
+void func_80ACBDFC(ObjAqua* this, GlobalContext* globalCtx) {
+    f32 temp = this->counter * 3.25f;
+    s32 pad;
+
+    this->alpha = (s32)(temp) + 10;
+    this->actor.shape.shadowAlpha = this->alpha;
+    this->unk_198 += 1000;
+    if (AQUA_HOT(&this->actor)) {
+        f32 temp_f2 = this->actor.scale.x * 10000.0f;
+
+        EffectSsBubble_Spawn(globalCtx, &this->actor.world.pos, temp_f2 * -0.5f, temp_f2, temp_f2,
+                             (Rand_ZeroOne() * 0.1f) + 0.03f);
+    }
+    if (this->counter <= 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBDFC.s")
+#endif
+
+void ObjAqua_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjAqua* this = THIS;
+    s32 pad;
+
+    if (this->counter > 0) {
+        this->counter -= 1;
+    }
+    this->actionFunc(this, globalCtx);
+    if (this->actor.update) {
+        if (this->actionFunc == func_80ACBC8C) {
+            Math_Vec3f_StepTo(&this->actor.scale, &D_80ACC308, 0.00006f);
+        } else if (this->actionFunc == func_80ACBD48) {
+            Math_Vec3f_StepTo(&this->actor.scale, &D_80ACC314, 0.00095f);
+        } else {
+            Math_Vec3f_StepTo(&this->actor.scale, &D_80ACC320, 0.0004f);
+        }
+        this->actor.velocity.y *= 0.9f;
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 12.0f, 4.0f, 0.0f, 5);
+        if (this->actionFunc != func_80ACBDFC) {
+            Collider_UpdateCylinder(&this->actor, &this->collider);
+            this->collider.dim.radius = this->actor.scale.x * 3000.0f;
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/ObjAqua_Update.s")
+void ObjAqua_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ObjAqua* this = THIS;
+    s32 framesTemp;
+    s32 pad;
+    s16 cameraTemp = func_800DFCDC(globalCtx->cameraPtrs[globalCtx->activeCamera]) + 0x8000;
```

Comment:
```suggestion
    s16 yaw = func_800DFCDC(globalCtx->cameraPtrs[globalCtx->activeCamera]) + 0x8000;
```
I've been working on camera and the name of `func_800DFCDC` is `Camera_GetCamDirYaw`, so it's returning a yaw (angle) instead of a camera